packages/kbn-securitysolution-alerts-grouping/src/components/types.ts

```diff
@@ -1,13 +1,17 @@
 /*
  * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
  * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
  */
 
-import type { GenericBuckets } from '../../../../common/search_strategy/common';
-
-export const DEFAULT_GROUPING_QUERY_ID = 'defaultGroupingQuery';
+// copied from common/search_strategy/common
+export interface GenericBuckets {
+  key: string | string[];
+  key_as_string?: string; // contains, for example, formatted dates
+  doc_count: number;
+}
 
 export const NONE_GROUP_KEY = 'none';
 
```

Comment:
I'm suggesting to make next changes to types:
make it generic to avoid deps to `alerts`, rename `stackByMultipleFields0` to `groupByMultipleFields0`, `groupsCount0` (which is not use anywhere) to `groupsNumber`:
```suggestion
export type RawBucket<T> = GenericBuckets & T;

/** Defines the shape of the aggregation returned by Elasticsearch */
export interface GroupingTableAggregation<T> {
  groupByMultipleFields0: {
    buckets?: Array<RawBucket<T>>;
  };
  groupsNumber: {
    value?: number | null;
  };
}

export type GroupingFieldTotalAggregation = Record<string, { value?: number | null }>;

export type FlattenedBucket<T> = Pick<RawBucket<T>, 'doc_count' | 'key' | 'key_as_string'> & {
  stackByMultipleFields1Key?: string;
  stackByMultipleFields1DocCount?: number;
};

```

Reply:
then we will have later `RawBucket<AlertGroupBucket>`

Reply:
eh?
```
export type RawBucket<T> = GenericBuckets &
  T & {
    stackByMultipleFields1?: {
      buckets?: GenericBuckets[];
      doc_count_error_upper_bound?: number;
      sum_other_doc_count?: number;
    };
  };
```

Reply:
I made a follow up issue for this https://github.com/elastic/kibana/issues/152491